Scripts/run-tests.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -o pipefail
+exitCode=0
+
+function checkStatus {
+    if [[ $exitCode == 0 && $1 != 0 ]]; then
+        exitCode=$1
+    fi
+}
+
+function uploadCodecov {
+    bash <(curl -s https://codecov.io/bash)
+}
+
```

Comment:
We should probably add some logic to set default values for `$IOS_SDK` and `$IOS_DESTINATION` here -- since the script will fail miserable if they aren't set.  I think this is pretty easy to do in bash:
```
IOS_SDK=${IOS_SDK:-iphonesimulator10.3}
```